src/plugins/FullScreenPlugin.js

```diff
@@ -0,0 +1,113 @@
+import { ButtonPlugin } from '../base-plugins/ButtonPlugin';
+import { Button } from '../ui-elements';
+
+/**
+ * A Springroll plugin to easily set up togglable fullscreen
+ */
+export class FullScreenPlugin extends ButtonPlugin {
+  
+  /**
+   *  Creates an instance of FullscreenPlugin
+   * 
+   * @param {string} targetElementSelector The selector for the element to be made fullscreen
+   * @param {string || HTMLElement } buttonSelector The selector for the button or the HTMLElement Button to be used to toggle fullscreen on the targetElement
+   */
+  constructor(buttonSelector) {
+    super({
+      name: 'fullScreen'
+    }); 
+
+    this._toggleButtons = [];
+    this.iFrame = null;
+
+    this.sendAllProperties = this.sendAllProperties.bind(this);
+
+    
+    this.toggleButton = document.querySelectorAll(buttonSelector);
+    this.toggleButton.forEach((button) => {
+      this._toggleButtons.push(new Button({
+        button: button,
+        onClick: this.toggleFullScreen.bind(this),
+        channel: 'fullScreen'
+      }));
+    });
+    
+
+    document.addEventListener('fullscreenchange',  () => {
+      this.sendProperty('fullScreen', this.isFullScreen);
+      
+      this._toggleButtons.forEach((button) => {
+        button.button.classList.toggle('--fullScreen');
+      });
+
+    });
+  }
+
+  /**
+   * @memberof FullScreenPlugin
+   */
+  init({ iframe }) {
+    this.iFrame = iframe;
+    // Handle the features request
+    this.client.on(
+      'features',
+      function($event) {
+        for (let i = 0; i < this.fullscreenElement; i ++) {
+          this._toggleButtons[i].displayButton($event.data);
+        }
+
+      }.bind(this)
+    );
+  }
+  /**
+  * @memberof FullScreenPlugin
+  */
+  start() {
+    this.client.on('loaded', this.sendAllProperties);
+    this.client.on('loadDone', this.sendAllProperties);
+  }
+
+  /**
+  *
+  * Sends initial fullScreen properties to the application
+  * @memberof FullScreenTogglePlugin
+  */
+  sendAllProperties() {
+    this.sendProperty(FullScreenPlugin.fullscreenKey, document.fullscreenElement != null ? 'true' : 'false');
+  }
+
+  /**
+   * Toggles fullscreen on this.iFrame. Must be from a user generated event
+   */
+  toggleFullScreen() {
+    if (!document.fullscreenElement) {
+      this.iFrame.requestFullscreen().then(() => {
+        this.sendProperty(FullScreenPlugin.fullscreenKey, document.fullscreenElement != null ? 'true' : 'false');
+      }).catch((err) => {
+        console.log(err);
+      });
+    } else {
+      document.exitFullscreen();
+      this.sendProperty(FullScreenPlugin.fullscreenKey, !document.fullscreenElement ? 'true' : 'false');
+    }
+  }
+
+  /**
+   * Returns true if there is a fullscreen element and false if not
+   * @returns { boolean } 
+   */
+  get isFullScreen() {
```

Comment:
This actually returns an object, which when sent via the postMessage stuff actually just gets turned into something like `[object Object]`. You should throw a `!!` in there to ensure that it will always be a boolean.